### PR TITLE
Improve Developer's guide

### PR DESCRIPTION
**Pull request by nqb**
# Description
- Add documentation conventions for writing Asciidoc
- Checklist for new guides
- Template for new guides

# Impacts
Embedded doc

# Delete branch after merge
YES

### CODE REVIEW

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 1]
```diff
@@ -57,6 +57,24 @@ To do so, the preferred way is using the `perldoc` command as follows:
   perldoc lib/pfconfig/cached.pm
 
 
+== Asciidoc documentation
```
Comment (extrafu) [inline docs/PacketFence_Developers_Guide.asciidoc]: Asciidoc documentation =>  AsciiDoc Documentation

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 2]
```diff
@@ -57,6 +57,24 @@ To do so, the preferred way is using the `perldoc` command as follows:
...
+
+=== Documentation conventions
```
Comment (extrafu) [inline docs/PacketFence_Developers_Guide.asciidoc]: Documentation conventions => Documentation Conventions

**Comment by nqb** [comment]
@extrafu, you can review.

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 3]
```diff
@@ -57,6 +57,305 @@ To do so, the preferred way is using the `perldoc` command as follows:
...
+|Element|Mark-up|Example rendered output
+
+|Application name
+a|No special markup
+a| The foo application.
+
+|Code blocks
+a|
+....
+[source,java]
```
Comment (extrafu) [inline docs/PacketFence_Developers_Guide.asciidoc]: I think we should use a Perl or Golang example. A Java example when there's no Java in PacketFence doesn't make sense.

Comment (nqb) [reply docs/PacketFence_Developers_Guide.asciidoc]: Done.

**Pull request merged**